packages/serverless-contracts/src/contracts/eventBridge/features/putEvent.ts

```diff
@@ -6,19 +6,37 @@ import { PutEventBuilderArgs, PutEventSideEffect } from '../types/putEvent';
 export const buildPutEvent =
   <Contract extends EventBridgeContract>(
     contract: Contract,
-    { eventBusName, source, eventBridgeClient }: PutEventBuilderArgs<Contract>,
+    {
+      eventBusName,
+      source,
+      eventBridgeClient,
+      throwOnFailure = true,
+    }: PutEventBuilderArgs<Contract>,
   ): PutEventSideEffect<Contract> =>
   async payload => {
     const event = {
       Detail: JSON.stringify(payload),
       DetailType: contract.eventType,
       Source: source,
-      EventBusName: eventBusName,
+      EventBusName:
+        typeof eventBusName === 'string' ? eventBusName : await eventBusName(),
     };
 
     const command = new PutEventsCommand({
       Entries: [event],
     });
 
-    await eventBridgeClient.send(command);
+    const { FailedEntryCount, Entries } = await eventBridgeClient.send(command);
+
+    if ((FailedEntryCount ?? 0) > 0 && throwOnFailure) {
+      const failedEntry = Entries?.[0];
+      console.error(
+        `Failed to send this event: ${JSON.stringify(event, null, 2)}`,
+      );
+      throw new Error(
+        `Failed to send event. Error: ${failedEntry?.ErrorMessage} (${failedEntry?.ErrorCode})`,
+      );
+    }
+
+    return { failedEntryCount: FailedEntryCount, entry: Entries?.[0] };
```

Comment:
This could be confusing if several entries fail. Maybe the error should at least contain all of them

Reply:
This is the put event (without "s") there is only one entry

Reply:
Oh you're right, my bad